drake/systems/frames/@CoordinateFrame/CoordinateFrame.m

```diff
@@ -68,17 +57,26 @@
         ind=ind+1;
       end
       if (nargin<4 || isempty(coordinates))
-        obj.coordinates=cellfun(@coordinateName,cell(dim,1),'UniformOutput',false);
+        coordinates=cellfun(@coordinateName,cell(dim,1),'UniformOutput',false);
       else
         typecheck(coordinates,'cell');
         sizecheck(coordinates,dim);
         for i=1:dim
           typecheck(coordinates{i},'char');
         end
-        obj.coordinates = {coordinates{:}}';
+        coordinates = {coordinates{:}}';
       end
+      
+      mex_ptr_args = cell(1,3);
+      [mex_ptr_args{:}] = CoordinateFrame.new(name,dim,coordinates);
+      obj = obj@DrakeMexPointer(mex_ptr_args{:});
+      obj.prefix = prefix;
     end
-    
+
```

Comment:
We should change the arguments to this function to `varargin`. Otherwise someone who was slicing into the coordinates list like this:

```
frame.coordinates(1:10)
```

will get a `Too many input arguments` error instead of the deprecation error.